File: DynamicKey/AgoraDynamicKey/php/test/RtmTokenBuilderTest.php
<?php
include "../src/RtmTokenBuilder.php";
include "TestTool.php";

$appID = "970CA35de60c44645bbae8a215061b33";
$appCertificate = "5CFd2fd1755d40ecb72977518be15d3b";
$account = "test_user";
$ts = 1111111;
$salt = 1;
$expiredTs = 1446455471;


$expected = "006970CA35de60c44645bbae8a215061b33IAAsR0qgiCxv0vrpRcpkz5BrbfEWCBZ6kvR6t7qG/wJIQob86ogAAAAAEAABAAAAR/QQAAEA6AOvKDdW";
$builder = new RtmTokenBuilder($appID, $appCertificate, $account);
$builder->setPrivilege(AccessToken::Privileges["kRtmLogin"], $expiredTs);
$builder->token->message->salt = $salt;
$builder->token->message->ts = $ts;
$result = $builder->buildToken();
assertEqual($expected, $result);


$expected = "006970CA35de60c44645bbae8a215061b33IABR8ywaENKv6kia6iUU6P54g017Bi6Ym9sIGdt9f3sLLYb86ogAAAAAEAABAAAAR/QQAAEA6ANkAAAA";
$builder = new RtmTokenBuilder($appID, $appCertificate, $account);
$builder->setPrivilege(AccessToken::Privileges["kRtmLogin"], 100);
$builder->token->message->salt = $salt;
$builder->token->message->ts = $ts;
$result = $builder->buildToken();
assertEqual($expected, $result);

?>

File: DynamicKey/AgoraDynamicKey/go/test/RtmTokenBuilder_test.go
package test

import (
	rtmtokenbuilder "RtmTokenBuilder"
	"testing"

	"../src/AccessToken"
)

func Test_RtmTokenBuilder(t *testing.T) {
	expected :=
		"006970CA35de60c44645bbae8a215061b33IAAsR0qgiCxv0vrpRcpkz5BrbfEWCBZ6kvR6t7qG/wJIQob86ogAAAAAEAABAAAAR/QQAAEA6AOvKDdW"

	appID := "970CA35de60c44645bbae8a215061b33"
	appCertificate := "5CFd2fd1755d40ecb72977518be15d3b"
	account := "test_user"
	expireTimestamp := uint32(1446455471)

	builder := rtmtokenbuilder.CreateRtmTokenBuilder(appID, appCertificate, account)
	builder.Token.Salt = uint32(1)
	builder.Token.Ts = uint32(1111111)
	builder.Token.Message[AccessToken.KLoginRtm] = expireTimestamp

	if result, err := builder.BuildToken(); err != nil {
		t.Error(err)
	} else {
		if result != expected {
			t.Error("Error ")
			t.Error(result)
		}
	}
}
